src/tiledb/cloud/udf.py

```diff
@@ -550,14 +550,26 @@ def unshare(name=None, namespace=None, async_req=False):
 """
 
 
-def delete(name, namespace, async_req=False):
+def delete(
```

Comment:
I believe this order change breaks the compatibility of the API? Users that are passing by position today would have breakage?

Reply:
I made the change to start keeping a more uniform style on those methods going forward.

What I can do is rename this method to `delete_new` and reinstate the previous `delete` method to work as expected with a deprecation warning.

Reply:
How about we add the uri parameter, to this function, deprecate the existing parameters and warn users. Then we can a avoid the `v2` function name, support compatibility for now and give us a clean future API?

Reply:
@ihnorton both deprecation approaches have their pros and cons.
We will highly value your input on which method to follow going forward (many methods will eventually be refactored)

Reply:
We could make the change but fall back to the existing (interpret as `name`) behavior if the argument doesn't start with `tiledb://` -- and print a deprecation warning.

Reply:
According to @ihnorton suggestion, I modified the method as shown here.
The `namespace` argument will be available for backwards compatibility until version `0.12.17` and the method will throw a deprecation warning if the `uri` is not valid.
FYI @Shelnutt2